mne/epochs.py

```diff
@@ -2052,6 +2052,115 @@ def _drop_log_stats(drop_log, ignore=('IGNORED',)):
     return perc
 
 
+def make_metadata(*, events, event_id, tmin, tmax, sfreq):
```

Comment:
I would not lead with `*`, it makes it harder to use and decreases the likelihood of what I think would be rare errors. In this specific case, this order follows that of Epochs already.

In the general case, I think the principle currently applied here would generalize to "never allow people to use positional arguments in new functions and methods, they should all start with `*`", which I don't think is what we want. I think we should use `*` sparingly to prevent bad errors where people can get in trouble mixing up `verbose` with something else like we did in `apply_baseline` for example.

Reply:
I wanted to make at least part of the params kw-only because this function is so new and we might want to change the keyword order as it evolves. Thought this would make for a more painless transition then. But I agree that making **all** params kw-only is probably a little bit too much.